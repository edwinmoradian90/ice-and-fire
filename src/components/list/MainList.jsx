import React from 'react'
import MainListItem from './MainListItem'
import { Loader } from '../loading/Loader'

function MainList(props) {
    const { data } = props.data
    console.log(props)
    return (
        <>
            {data
                ? Object.keys(data).map((item, i) => (
                    <li key={i}><MainListItem {...props} item={item} /></li>
                ))
                :
                (
                    <Loader />
                )
            }
        </>
    )
}

export default MainList