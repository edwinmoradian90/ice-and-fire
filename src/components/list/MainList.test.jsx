import React from 'react';
import renderer from 'react-test-renderer';
import MainList from './MainList';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Main List', () => {
    const url = 'https://anapioficeandfire.com/api';
    const data = {
        data: {
            books: `${url}/books`,
            characters: `${url}/characters`,
            houses: `${url}/houses`,
        },
    };
    it('renders properly', () => {
        const component = renderer.create(
            <Router>
                <MainList data={data} />
            </Router>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});