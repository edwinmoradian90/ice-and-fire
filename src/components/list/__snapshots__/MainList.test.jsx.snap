// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main List renders properly 1`] = `
Array [
  <div>
    <a
      className="sc-AxjAm irMvbk listItem"
      href="/books"
      onClick={[Function]}
    >
      <div
        className="sc-AxhUy ilDfEv itemTitle"
      >
        books
      </div>
    </a>
  </div>,
  <div>
    <a
      className="sc-AxjAm irMvbk listItem"
      href="/characters"
      onClick={[Function]}
    >
      <div
        className="sc-AxhUy ilDfEv itemTitle"
      >
        characters
      </div>
    </a>
  </div>,
  <div>
    <a
      className="sc-AxjAm irMvbk listItem"
      href="/houses"
      onClick={[Function]}
    >
      <div
        className="sc-AxhUy ilDfEv itemTitle"
      >
        houses
      </div>
    </a>
  </div>,
]
`;
